v2/yarpcgrpc/integration_test.go

```diff
@@ -21,521 +21,391 @@
 	}
-	defer func() {
-		if err != nil {
-			err = multierr.Append(err, inbound.Stop())
-		}
-	}()
-
-	var clientConn *grpc.ClientConn
 
-	clientConn, err = grpc.Dial(listener.Addr().String(), newDialOptions(dialOptions).grpcOptions()...)
-	if err != nil {
-		return nil, err
+	inbound := options.Inbound
+	if inbound == nil {
+		inbound = &Inbound{}
 	}
-	keyValueClient := examplepb.NewKeyValueClient(clientConn)
-
-	chooser := peer.NewSingle(hostport.Identify(listener.Addr().String()), trans.NewDialer(dialOptions...))
-	outbound := trans.NewOutbound(chooser, outboundOptions...)
 
-	if err := outbound.Start(); err != nil {
+	inbound.Addr = "127.0.0.1:0"
+	inbound.Router = yarpctest.NewFakeRouter(options.Procedures)
+	if err := inbound.Start(context.Background()); err != nil {
 		return nil, err
 	}
-	defer func() {
-		if err != nil {
-			err = multierr.Append(err, outbound.Stop())
-		}
-	}()
 
-	caller := "example-client"
-	service := "example"
-	clientConfig := clientconfig.MultiOutbound(
-		caller,
-		service,
-		transport.Outbounds{
-			ServiceName: caller,
-			Unary:       outbound,
-		},
-	)
-	keyValueYARPCClient := examplepb.NewKeyValueYARPCClient(clientConfig)
+	outbound := options.Outbound
+	if outbound == nil {
+		outbound = &Outbound{}
+	}
+	outbound.Dialer = dialer
+	outbound.URL = parseURL("http://" + inbound.Listener.Addr().String())
 
-	contextWrapper := grpcctx.NewContextWrapper().
-		WithCaller("example-client").
-		WithService("example").
-		WithEncoding(string(protobuf.Encoding))
+	client := yarpc.Client{
+		Service: "test-service",
+		Caller:  "test-caller",
+		Unary:   outbound,
+	}
 
 	return &testEnv{
-		Caller:              caller,
-		Service:             service,
-		Transport:           trans,
-		Inbound:             inbound,
-		Outbound:            outbound,
-		ClientConn:          clientConn,
-		ContextWrapper:      contextWrapper,
-		ClientConfig:        clientConfig,
-		Procedures:          procedures,
-		KeyValueGRPCClient:  keyValueClient,
-		KeyValueYARPCClient: keyValueYARPCClient,
-		KeyValueYARPCServer: keyValueYARPCServer,
+		Inbound:  inbound,
+		Outbound: outbound,
+		Dialer:   dialer,
+		Client:   client,
 	}, nil
 }
 
-func (e *testEnv) Call(
-	ctx context.Context,
-	methodName string,
-	message proto.Message,
-	encoding transport.Encoding,
-	headers transport.Headers,
-) (*transport.Response, error) {
-	data, err := proto.Marshal(message)
-	if err != nil {
-		return nil, err
-	}
-	ctx, cancel := context.WithTimeout(ctx, testtime.Second)
-	defer cancel()
-	return e.Outbound.Call(
-		ctx,
-		&transport.Request{
-			Caller:   e.Caller,
-			Service:  e.Service,
-			Encoding: encoding,
-			Procedure: procedure.ToName(
-				"uber.yarpc.internal.examples.protobuf.example.KeyValue",
-				methodName,
-			),
-			Headers: headers,
-			Body:    bytes.NewReader(data),
-		},
+func (e *testEnv) Close() error {
+	return multierr.Combine(
+		e.Dialer.Stop(context.Background()),
+		e.Inbound.Stop(context.Background()),
```

Comment:
Do we need to stop any of the other dependencies in the `testEnv` here?

Reply:
nope